PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,261 @@
+"""
+
+from selenium import webdriver
+from selenium.webdriver.common.action_chains import ActionChains
+from selenium.webdriver.common.keys import Keys
+from selenium.webdriver.support.ui import Select
+from selenium.common.exceptions import NoSuchElementException
+import sys
+import csv
+from bs4 import BeautifulSoup
+import time
+
+
+def main():
+    """Login De Anza portal."""
+    driver = webdriver.Chrome()
+    login_myportal()
+    # The way to judge is that the left menu can be found in the interface, and the menu style has list-group-item)
+    while True:
+        try:
+            driver.find_element_by_class_name("list-group-item")
+        except NoSuchElementException:
+            time.sleep(1)
+            print("wait for seconds...")
+            continue
+        break
+
+    # Course search page from homepage after login
+    openSearchPage(driver)
+
+    # choose course
+    select = driver.find_element_by_tag_name("select")  # Because the page has only one drop-down box
+
+    # Select specified course
+    downlist = Select(select)  # a Select object
+    downlist.select_by_value("202022")  # Element positioning through select objects, value positioning (value of option)
+    # search submit button
+    submitSearch(driver)
+    # submit
+    submitClick(driver)
+
+    # Wait for a while to make sure the page loads
+    waitUtilPageLoaded(driver)
+    # Go to the advanced options page and start filling in various search terms
+    fillAdvanceSearch(driver)
+    # Save searched courses
+    html = saveResult(driver)  # see method
+    courseList = []
+    get_contents(courseList, html)  #Extract course information from html
+    save_contents(courseList) #Save course information as a csv file
+
+
+def submitSearch(driver):
+    """Search sumbit button."""
+    ipts = driver.find_elements_by_tag_name("input")
+    submit = ""
+    for ipt in ipts:
+        if "submit".lower() == ipt.get_attribute("type") and ipt.is_enabled() and ipt.is_displayed():
+            submit = ipt
+            break
+    if not submit:
+        print("No submit button found！")
+        return
+    submit.click()
+    time.sleep(2)  # Waiting for completion
+
+
+def submitClick(driver):
+    """Locate submit button and click."""
+    # Locate and click the Submit button
+    ipts = driver.find_elements_by_tag_name("input")
+    submit = ""
+    for ipt in ipts:
+        if "submit".lower() == ipt.get_attribute("type") and \
+                "Advanced Search" == ipt.get_attribute("value"):
+            submit = ipt
+            break
+
+    if not submit:
+        print("Advanced Search button not found！")
+        return
+
+    submit.click()
+
+
+# login
+def login_myportal(driver):
+    """Open myportal website and login."""
+    # Open the browser (The true url is：https://myportal.fhda.edu/）
+    driver.get("https://myportal.fhda.edu/")
+    # input username
+    username = int(input("Please enter your user name:  "))
```

Comment:
try catch `ValueError`

Reply:
Due to the 250 lines control of total codes, I can't add any other code.